### PR TITLE
Add siddhi-store-cosmosdb source files

### PR DESCRIPTION
**Pull request by tdia2082**
## Purpose
Add siddhi-store-cosmosdb source files.

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -178,15 +178,23 @@
    APPENDIX: How to apply the Apache License to your work.
 
       To apply the Apache License to your work, attach the following
+<<<<<<< HEAD
```
Comment (ramindu90) [inline LICENSE]: @Tikiri16 we can remove these extra lines

File: component/src/main/java/io/siddhi/extension/store/cosmosdb/CosmosDBEventTable.java [hunk 2]
```diff
@@ -0,0 +1,701 @@
...
+
+import static io.siddhi.core.util.SiddhiConstants.ANNOTATION_STORE;
+
+/**
+ * Class representing CosmosDB Event Table implementation.
+ */
+@Extension(
+        name = "cosmosdb",
+        namespace = "store",
+        description = "Using this extension a Cosmos DB Event Table can be configured to persist events " +
```
Comment (ramindu90) [inline component/src/main/java/io/siddhi/extension/store/cosmosdb/CosmosDBEventTable.java]: @Tikiri16 shall we rephrase this as "This extension will be used to execute Create, Update, Delete, Find and Contains operations on document collections in Cosmos DB"

File: component/src/test/java/io.siddhi.extension.store.cosmosdb/DeleteFromCosmosTableTest.java [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    }
+
+    @Test
+    public void deleteFromCosmosTableTest1() throws InterruptedException {
+        log.info("deleteFromCosmosTableTest1 - " +
+                "Delete an event of a CosmosDB table successfully");
+
+        String collectionLink = String.format("/dbs/%s/colls/%s", database, "FooTable");
+        CosmosTableTestUtils.dropCollection(uri, key, collectionLink);
+
```
Comment (ramindu90) [inline component/src/test/java/io.siddhi.extension.store.cosmosdb/DeleteFromCosmosTableTest.java]: @Tikiri16 shall we remove those extra spaces in all the test files?

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+        <ow2.jta.spec.version>1.0.12</ow2.jta.spec.version>
+        <commons.io.version>1.3.2</commons.io.version>
+        <osgi.test.util.version>5.1.1</osgi.test.util.version>
+        <slf4j.version>1.7.12</slf4j.version>
+
+        <bind.host.name>localhost</bind.host.name>
+
+        <log4j.version>1.2.17.wso2v1</log4j.version>
+        <testng.version>6.9.10</testng.version>
+        <mysql.connector.version>5.1.38</mysql.connector.version>
```
Comment (ramindu90) [inline pom.xml]: @Tikiri16 shall we remove these since its not needed? please remove all the dependencies which uses these if not needed as well.

File: pull_request_template.md [hunk 5]
```diff
@@ -1,52 +0,0 @@
-## Purpose
```
Comment (ramindu90) [inline pull_request_template.md]: @Tikiri16 Why is this removed?

**Review by ramindu90 — CHANGES_REQUESTED** [review]

File: component/src/main/java/io/siddhi/extension/store/cosmosdb/CosmosDBEventTable.java (hunk 2 above)
Comment (tdia2082) [reply component/src/main/java/io/siddhi/extension/store/cosmosdb/CosmosDBEventTable.java]: Changed.

File: component/src/test/java/io.siddhi.extension.store.cosmosdb/DeleteFromCosmosTableTest.java (hunk 3 above)
Comment (tdia2082) [reply component/src/test/java/io.siddhi.extension.store.cosmosdb/DeleteFromCosmosTableTest.java]: removed.

File: pom.xml (hunk 4 above)
Comment (tdia2082) [reply pom.xml]: Done.

File: pull_request_template.md (hunk 5 above)
Comment (tdia2082) [reply pull_request_template.md]: That was by mistake, will add that.

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+            <dependency>
+                <groupId>org.jacoco</groupId>
+                <artifactId>org.jacoco.ant</artifactId>
+                <version>${jacoco.ant.version}</version>
+            </dependency>
+
+            <dependency>
+                <groupId>org.apache.commons</groupId>
+                <artifactId>commons-lang3</artifactId>
+                <version>3.3.2</version>
```
Comment (ramindu90) [inline pom.xml]: @Tikiri16 shall we move these to properties section?

Comment (tdia2082) [reply pom.xml]: Moved.

**Review by ramindu90 — APPROVED** [review]

**Review by ramindu90 — APPROVED** [review]

**Review by tishan89 — APPROVED** [review]
